cpp/include/cugraph_c/algorithms.h

```diff
@@ -219,6 +222,36 @@ cugraph_error_code_t cugraph_bfs(
   cugraph_paths_result_t** result,
   cugraph_error_t** error);
 
+/**
+ * @brief     Perform single-source shortest-path to compute the minimum distances
+ *            (and predecessors) from the source vertex.
+ *
+ * This function computes the distances (minimum edge weight sums) from the source
+ * vertex. If @p predecessors is not NULL, this function calculates the predecessor of each
+ * vertex (parent vertex in the breadth-first search tree) as well.
+ *
+ * @param [in]  handle       Handle for accessing resources
+ * @param [in]  graph        Pointer to graph
+ * @param [in]  source       Source vertex id
+ * @param [in]  cutoff       Maximum edge weight sum to consider
+ * @param [in]  do_expensive_check A flag to run expensive checks for input arguments (if set to
+ * `true`).
+ * @param [in]  compute_predecessors A flag to indicate whether to compute the predecessors in the
+ * result
+ * @param [out] result       Opaque pointer to paths results
+ * @param [out] error        Pointer to an error object storing details of any error.  Will
+ *                           be populated if error code is not CUGRAPH_SUCCESS
+ * @return error code
+ */
+cugraph_error_code_t cugraph_sssp(const cugraph_resource_handle_t* handle,
+                                  cugraph_graph_t* graph,
```

Comment:
C api will transpose the graph if the input is opposite from what the algorithm requires.  This will modify the graph.

Reply:
Would it be better to set/return an error in that case?  I'm envisioning a case where we or a user has a bug that isn't transposing data if an algo requires it, it gets silently transposed anyway, then we wonder why our benchmarks are slower than expected.

Reply:
Fair question.  I think we've discussed this (or else I had this argument with myself... entirely possible).

From an API perspective, we have two choices:
 * Do what the user is asking us to do regardless of the performance ramifications.  The documentation should provide sufficient detail for the user to correct their bad choices.
 * Require the user to understand the performance ramifications and make all of the right calls explicitly

The C++ layer has always opted for the latter and I believe it should.  Thus far I have opted for the former in the C API.  Now that @rlratzel is building the `python` layer it would make sense for us to have this conversation.  Perhaps the `python` layer should worry about providing the intelligence and the C layer should (like the C++ layer) simply fail fast if the input is in the wrong orientation.